### PR TITLE
ARROW-13230: [Docs][Python] Add CSV writer docs

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

File: docs/source/cpp/csv.rst [hunk 1]
```diff
@@ -69,6 +69,44 @@ A CSV file is read from a :class:`~arrow::io::InputStream`.
...
+       std::shared_ptr<arrow::io::OutputStream> output = ...;
+       auto write_options = arrow::csv::WriteOptions::Defaults();
+       auto maybe_writer = arrow::csv::MakeCSVWriter(output, schema, options);
+       if (!maybe_writer.ok()) {
+           // Handle writer instantiation error...
+       }
+       std::shared_ptr<arrow::ipc::RecordBatchWriter> writer = *maybe_writer;
+
+       // Write batches...
+       if (!writer->WriteRecordBatch(*batch).ok()) {
```
Comment (emkornfield) [inline docs/source/cpp/csv.rst:98]: I don't feel too strongly especially if this is consistent with other docs, but I think using status macros might make sense here.

File: docs/source/python/csv.rst [hunk 2]
```diff
@@ -64,6 +65,16 @@ with the file path you want to read from::
    3       23.68  3.31    Male     No  Sun  Dinner     2
    4       24.59  3.61  Female     No  Sun  Dinner     4
 
+To write CSV files, just call :func:`write_csv` with a
```
Comment (emkornfield) [inline docs/source/python/csv.rst:68]: it might be worth a note that not all arrow types are supported yet.

File: python/setup.py [hunk 3]
```diff
@@ -108,6 +108,7 @@ def run(self):
                       'namespace of boost (default: boost)'),
                      ('with-cuda', None, 'build the Cuda extension'),
                      ('with-flight', None, 'build the Flight extension'),
+                     ('with-hdfs', None, 'build the HDFS extension'),
```
Comment (emkornfield) [inline python/setup.py:111]: this seems unrelated?

**Comment by emkornfield** [comment]
Thanks for adding this.  LGTM with a few nits.

File: python/setup.py (hunk 3 above)
Comment (lidavidm) [reply python/setup.py]: Removed (sorry, it bugs me when I have to remember which of env var/CLI argument to use for each component).

File: docs/source/python/csv.rst (hunk 2 above)
Comment (lidavidm) [reply docs/source/python/csv.rst:68]: Added a note below (and in the C++ docs too).

File: docs/source/cpp/csv.rst (hunk 1 above)
Comment (lidavidm) [reply docs/source/cpp/csv.rst:98]: Looks like we generally don't use the macros, nor do we use ValueOrDie - it is a bit messy here though, I'll give you that.

**Comment by emkornfield** [comment]
+1, will try to merge on green.
